src/lib/journey/journey.store.ts

```diff
@@ -84,52 +69,72 @@ export async function initialize(journey?: string): Promise<InitObject> {
     const previousPayload = prevStep?.payload;
     let nextStep: StepTypes;
 
-    step.set(prevStep);
+    set({
+      completed: false,
+      error: null,
+      loading: true,
+      step: prevStep,
+      successful: false,
+      response: null,
+    });
+
     try {
       /**
        * Initial attempt to retrieve next step
        */
-      nextStep = await FRAuth.next(get(step as Writable<FRStep>), options);
+      nextStep = await FRAuth.next(prevStep as FRStep, options);
     } catch (err) {
       console.error(`Next step request | ${err}`);
 
       /**
        * Setup an object to display failure message
        */
       nextStep = new FRLoginFailure({
-        message: 'Unknown request failure'
+        message: 'Unknown request failure',
       });
     }
 
     if (nextStep.type === StepType.Step) {
-      step.set(nextStep);
-      submittingForm.set(false);
+      // Iterate on a successful progression
+      stepNumber = stepNumber + 1;
+
+      set({
+        completed: false,
+        error: null,
+        loading: false,
+        step: nextStep,
+        successful: false,
+        response: null,
+      });
     } else if (nextStep.type === StepType.LoginSuccess) {
-      // User is authenticated, now call for OAuth tokens
-      console.log('Calling OAuth flow.');
-      getOAuth();
-      step.set(nextStep);
-      submittingForm.set(false);
+      set({
+        completed: true,
+        error: null,
+        loading: false,
+        step: null,
+        successful: true,
+        response: nextStep.payload,
+      });
     } else if (nextStep.type === StepType.LoginFailure) {
```

Comment:
On login failure it does not look like we "reset" the store as we do for login success or next step; Silly question but is this intentional?

Reply:
Ok I think this is intentional since we want to capture the failure state for debugging purposes, however nothing seems to call reset() from within this file. Again I'm probably being silly, but want to check that we haven't forgotten to reset the state if there is a failure then the user tries again.

Reply:
Ok, forget what I'm saying - I can see further down in **_journey.svelte_** that **reset()** is being called. Ignore me!

Reply:
Just to make sure we are on the same page ... When we encounter a failure, we save off the failure message from the original request, and re-request the journey with no previous step to get a fresh set of callbacks. We then [`set` the new step with the old form failure message](https://github.com/cerebrl/forgerock-web-login-framework/pull/9/files#diff-e01603033c52bf6a043d215ff082252220802c8f537131e59b6fc3eb8de26cd3R223) on the store in order to communicate this failure to the user without presenting them a deadend. This is the main way of "resetting" the store for failures.

Now, the is a way to reset the store manually, and that's best for unrecoverable failures. This is the `reset` you may [see in the `journey.svelte` file](https://github.com/cerebrl/forgerock-web-login-framework/pull/9/files#diff-38f4b155f2dae7068eb3358d8b505a4b5730a0d12bc21bbec14ab47f89ba6bceR16). That happens when we failed on the first attempt and then something went wrong on the second attempt as well. To prevent looping, we bail out and show a UI that requires the user to restart the flow from scratch.

Reply:
Ok this makes more sense now - so we want to present feedback from the previous failure whilst presenting a new attempt to authenticate. Got it. Thanks